utils/s3/credentials_providers/config_file_aws_credentials_provider.hh

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+#include "aws_credentials_provider.hh"
+
+namespace aws {
+
+/**
+ * Reads credentials form yaml file as described in https://github.com/scylladb/scylladb/blob/master/docs/dev/object_storage.md
+ */
+
+class config_file_aws_credentials_provider final : public aws_credentials_provider {
+public:
+    explicit config_file_aws_credentials_provider(const std::string& _creds_file);
+    [[nodiscard]] seastar::future<s3::aws_credentials> get_aws_credentials() override;
+    [[nodiscard]] const char* get_name() const override { return "config_file_aws_credentials_provider"; }
+
+protected:
+    [[nodiscard]] bool is_time_to_refresh() const override { return false; }
```

Comment:
`return false;` means there's going to be no way to update config file and send HUP to scylla to re-read one? IIRC some test in object_store/ suite relies on it

Reply:
In client's `update_config` I invalidate creds so on next request it will be renewed
https://github.com/scylladb/scylladb/pull/21830/files#diff-28c1d03a5facd99f25f466e6bae94c0fab2bb8845c0538dd489db7832505e884R107